src/textual/app.py

```diff
@@ -227,13 +221,28 @@ def run(
             console (Console, optional): Console object. Defaults to None.
             screen (bool, optional): Enable application mode. Defaults to True.
             driver (Type[Driver], optional): Driver class or None for default. Defaults to None.
+            loop (): Event loop to run the application on. If not specified, uvloop will be used.
         """
 
         async def run_app() -> None:
             app = cls(screen=screen, driver_class=driver, **kwargs)
             await app.process_messages()
 
-        asyncio.run(run_app())
+        if loop:
+            asyncio.set_event_loop(loop)
+        else:
+            try:
```

Comment:
I think this installs uvloop globally. Perhaps it would be better to create a instance of the UVLoop?

https://uvloop.readthedocs.io/user/index.html

Reply:
Probably better yeah, done